stdlib/SparseArrays/src/sparsevector.jl

```diff
@@ -1935,6 +1935,16 @@ julia> dropzeros(A)
 """
 dropzeros(x::SparseVector; trim::Bool = true) = dropzeros!(copy(x), trim = trim)
 
+function copy!(dst::SparseVector, src::SparseVector)
+    dst.n == src.n || throw(ArgumentError("Vector should have the same length for copy!"))
+    copy!(dst.nzval, src.nzval)
+    copy!(dst.nzind, src.nzind)
+    return dst
+end
+
+function copy!(dst::SparseVector, src::AbstractVector)
+    copy!(dst, sparse(src))
```

Comment:
This seems too inefficient for a library function, as it creates an uncessary temporary allocation... 
I saw that in the issue you suspect implementing `empty!` for sparse vectors would be too inefficient: would you mind comparing the performance of that (using `empty!` and the default `copy!` implementation) versus your implementation of `copy!(dst::SparseVector, src::SparseVector)` above?

Reply:
I think that empty! cannot be really be implemented as long as `SparseVector` is immutable. Or am I misunderstanding?

Reply:
You are totally right, sorry again!

Reply:
Still, I'm hoping for a more efficient implementation here. That could be left for as a future optimization though.

Reply:
I hear you. I just pushed a more efficient version by cannibalizing dense2sparsevec. I haven't thoughly tested or benchmarked it yet though. But let me know if you think it is more or less what you had in mind.